BUILD.md

```diff
@@ -111,6 +113,27 @@ Step by step procedure:
    * Compile the whole project by selecting the `Build` -> `Build Solution`
      menu entry.
 
+ * On Windows (cmake):
+   * Create a `build` directory, which will contain all compile FreeOrion
+     build artifacs.
+   * Change into the `build` directory on the command line.
+   * Execute cmake to generate Makefiles:
+
+     ```
+     cmake .. -G "Visual Studio 15 2017"
+     ```
+   * Compile the whole project by calling `MSBuild.exe -p:Configuration=Release`
+     within the build directory. In case you want to utilize multiple CPU
+     cores, you can add the `-m` option to the command.
+   * Alternatively, you can compile the project by the Visual Studio GUI.
+   * Change into the `Release` directory, which contains the executables.
+   * Create a symbolic link to the data directory inside the build directoy
+     by invoking:
+
+     ```
+     ln -s ../../default .
```

Comment:
there is no `ln` command in the standard windows command prompt

Reply:
might be better to do what the existing MSVC project files do, which is to make the output file eg. `../../FreeOrion.exe`.

Reply:
I think CMake build is better to be out-of-source. I have updated the CMakeList.txt such that the `default`  directory is copied to the directory containing FreeOrion.exe

Reply:
copying the whole contents of `default` might not be the best default thing to do, no pun intended, as it is not trivially small on disk...

Reply:
Seems like symlink in CMake is not working properly in Windows. 

An easy "fixed" on this issue is to modify the build instruction, asking people to run the exe in the root directory of the source as `./build/Release/FreeOrion.exe`. Then it can detect the `default` folder automatically.

Honestly, 100 Mb is not that big in modern computers, and directly copying the `default` directory helps creating multiple source-independent builds.

If you think it is better to create symlink, I guess I can call some powershell/cmd commands to make the symlink.

Which one do you prefer?

Reply:
I don't really know what's generally preferred and don't really care for myself as I intend to keep using the existing project files and I find it works fine to put the binaries in the root directory of the source tree.

An alternative might be to use the `--resource.path` command line option?

Reply:
`--resource.path` works partially, but it seems like more parameters (e.g. audio) are required to start the game normally. For some reasons the other files can't be found automatically even if the `default` directory is passed to `--resource.path`.

If you have no strong preference, I will follow the first option - run `./build/Release/FreeOrion.exe`  at the source root, at least I know a few open source projects are doing in this way. The `Build.md` instruction has already been updated.